### PR TITLE
Add Java gRPC skeleton code for the parser

### PR DESCRIPTION
**Pull request by tonyyang-svail**
Part of solving https://github.com/sql-machine-learning/sqlflow/issues/1389.

### CODE REVIEW

File: java/parser/src/main/proto/Parser.proto [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  // The sql_program represents a SQL program that contains possible more than
+  // one SQL statements. For example, `select * from my_table; select * from another_table to train ...`.
+  string sql_program = 2;
+}
+
+message ParserResponse {
+  // The sql_statements represent part of the sql_program accepted by the parser.
+  // For example, if the ParserRequest is `select 1; select 2 to train ...`.
+  // The ParserResponse contains two statements, `select 1;` and `select 2 `
+  repeated string sql_statements = 1;
```
Comment (typhoonzero) [inline java/parser/src/main/proto/Parser.proto:32]: Do we need to return a struct like:

https://github.com/sql-machine-learning/sqlflow/blob/129be40d6e2627a30e7f06aaa8604d872542b6c4/pkg/parser/sqlflow_parser.go#L28-32

Comment (tonyyang-svail) [reply java/parser/src/main/proto/Parser.proto:32]: Yes, the API should be consistent with 
https://github.com/sql-machine-learning/sqlflow/blob/129be40d6e2627a30e7f06aaa8604d872542b6c4/pkg/parser/sqlflow_parser.go#L113

File: Dockerfile [hunk 2]
```diff
@@ -87,6 +87,7 @@ mv $GOPATH/bin/sqlflowserver /usr/local/bin && \
 mv $GOPATH/bin/repl /usr/local/bin && \
 (cd python/couler && python setup.py -q install) && \
 cd java/parser && \
+protoc --java_out=src/main/java --grpc-java_out=src/main/java/ --proto_path=src/main/proto/ src/main/proto/Parser.proto && \
```
Comment (wangkuiyi) [inline Dockerfile:90]: Is this line duplicated with the newly added line to test/java.sh?

Comment (wangkuiyi) [reply Dockerfile:90]: If so, it seems that we only need the line here, but not test/java.sh?

**Review by wangkuiyi — APPROVED** [review]

File: Dockerfile (hunk 2 above)
Comment (tonyyang-svail) [reply Dockerfile:90]: > Is this line duplicated with the newly added line to test/java.sh?

They are not duplicated. The line here is used for building the .jar file in the release image. The line in test/java.sh is used for testing.

This logic is consistent with having two `go generate ./...`s. One in Dockerfile, one in `scripts/test/units.sh`.

**Pull request merged**